src/test/java/com/scottlogic/AuthorPostSorterTest.java

```diff
@@ -0,0 +1,86 @@
+package com.scottlogic;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AuthorPostSorterTest {
+
+    @Test
+    void sort_withNull_returnsNull() {
+        List<UserPost> actual = null;
+        actual = new AuthorPostSorter().sort(actual, SortOrder.ASC);
```

Comment:
In general you want to avoid side effects and mutation inside tests. Instead of setting `actual` to be `null` and then also setting it to be the sorted version having a variable that is `initialList` or something similar makes the tests significantly clearer. The style of testing is different to normal coding as it is all about maximizing readability rather than efficiency as they will be run a finite number of times, this means a slight efficiency loss doesn't have much impact.